extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/CdcSinks.java

```diff
@@ -0,0 +1,270 @@
+            }
+            return valueFn.apply(record);
+        };
+    }
+
+    private static class CdcSinkProcessor<K, V> extends UpdateMapWithMaterializedValuesP<ChangeRecord, K, V> {
+
+        private Sequences<K> sequences;
+
+        CdcSinkProcessor(
+                @Nonnull HazelcastInstance instance,
+                @Nonnull String map,
+                @Nonnull FunctionEx<? super ChangeRecord, ? extends K> keyFn,
+                @Nonnull FunctionEx<? super ChangeRecord, ? extends V> valueFn
+        ) {
+            super(instance, map, keyFn, valueFn);
+        }
+
+        @Override
+        public void init(@Nonnull Outbox outbox, @Nonnull Context context) {
+            super.init(outbox, context);
+
+            HazelcastProperties properties = new HazelcastProperties(context.jetInstance().getConfig().getProperties());
+            int expiration = properties.getSeconds(SEQUENCE_CACHE_EXPIRATION_SECONDS);
+            this.sequences = new Sequences<>(expiration);
+        }
+
+        @Override
+        protected boolean shouldBeDropped(K key, ChangeRecord item) {
+            long timestamp = getTimestamp(item);
+
+            ChangeRecordImpl recordImpl = (ChangeRecordImpl) item;
+            long sequencePartition = recordImpl.getSequencePartition();
+            long sequenceValue = recordImpl.getSequenceValue();
+
+            boolean isNew = sequences.update(key, timestamp, sequencePartition, sequenceValue);
+            return !isNew;
+        }
+
+        private static long getTimestamp(ChangeRecord item) {
+            try {
+                return item.timestamp();
+            } catch (ParsingException e) {
+                //use current time, should be good enough for cache expiration purposes
+                return System.currentTimeMillis();
+            }
+        }
+
+    }
+
+    private static final class Sequences<K> {
+
+        private static final int INITIAL_CAPACITY = 64 * 1024;
+        private static final float LOAD_FACTOR = 0.75f;
+
+        private final LinkedHashMap<K, long[]> sequences;
+
+        public Sequences(int expiration) {
+            sequences = new LinkedHashMap<K, long[]>(INITIAL_CAPACITY, LOAD_FACTOR, true) {
+                @Override
+                protected boolean removeEldestEntry(Map.Entry<K, long[]> eldest) {
+                    long age = eldest.getValue()[2];
+                    return System.currentTimeMillis() - age > expiration;
+                }
+            };
+        }
+
+        boolean update(K key, long timestamp, long partition, long value) {
+            long[] prevSequence = sequences.get(key);
+            if (prevSequence == null) { //first observed sequence for key
+                sequences.put(key, new long[] {partition, value, timestamp});
+                return true;
+            } else {
+                prevSequence[2] = timestamp;
+                if (prevSequence[0] != partition) { //sequence partition changed for key
```

Comment:
can this happen?

Reply:
It should be possible, for example when you have a setup where the Debezium connector is able to switch from one DB to another. Is also the cause of simple numeric value not being enough as a sequence number.